### DISCUSSION TITLE
Sensor to ADC pyboard 1.1v protection advice

### DISCUSSION THREAD

**Discussion by w5im** [micropython/micropython#14221] [STM32 / Pyboard] [answered]
hello all,

am making a small project of DIY EMG sensor and seems to be good, I need to connect to the ADC of the pyboard 1.1v, just want to confirm that it is safe, powering the circuit with a couple of batteries (total 4.4v) the signal is rectified then last stage am using AD706 Op-amp for amplification, as testing the output never goes beyond 3V MAX (I think AD706 max output "Vss -1.2v"). I know that the ADC is 3.3v tolerance, so kindly I have many questions :
lets first: do I have to make some ADC input protection? 
if yes, then I have alot of discussions about that, what to do or not ... thanks in advance 😄 

![output](https://github.com/micropython/micropython/assets/165560453/3b5ac3f4-5135-4574-a6b8-831fa9c722e2)

**Comment by peterhinch** [comment]
You'll need to give us more information to comment. What voltage are +V and -V with respect to gnd? The circuit should be designed so that, while initialising and even under fault conditions, the voltage presented to the ADC cannot go below gnd or above 3.3V.

**Reply by w5im** [reply]
Hi peterhinch, thank you for replying.
The supply 6 AAA batteries the ground at the midpoint so it is around +4.4/- 4.4v.
That only for now for easy testing also with ( 4 AAA = 2.4V).
Am thinking about using one 9v batteries with voltage divider + capacitors or voltage regulator .
Thinking about what you said, I did not test the output while am connecting the 3 pins +-V and ground , it might go under ground but I guess will never go beyond 3.3v , I will test and update it .
But for now let assume I will connected the circuit to ADC after the batteries is ON and ready .
While testing it never went under ground, I can keep +0.5v offset .

The circuit simply cascaded of Op-amps , filters and precision rectifier, I will re-arrange the circuit to look more clear if need it ?
Am thinking should I worry about getting the opamp damage and shorting 4.4v to ADC , even they work at max 15v , or that would be so rare

**Reply by peterhinch** [reply, marked as answer]
There are CMOS op-amps available whose output can go rail-to-rail and can run on 3.3V (e.g. TLV2470 series). Run one of these on 3.3V and the output cannot possibly break the ADC.

**Reply by w5im** [reply]
am only using AD620 and AD706 they both can work with 3.3v, but the output will never be at 3V it will be around 2.2V.
am not experienced with ADC and how it is connects in practice, what I understand I will be losing some ADC range " reading resolution " , would be much better if I keep it, right? is it worth it?

**Reply by peterhinch** [reply]
You maximise accuracy if the voltage swing matches the ADC input range.

**Reply by w5im** [reply]
Hi again,
I think when you mentioned TLV2470, you meant to use it as a single supply 3.3v, I have been thinking since then.
I guess this is the best option to use, the funny thing is I could not find an OP-amp that fit my requirement until I reviewed your comment XD
I was looking for (low single supply, uV Vos, DIP package) even thou I was using: mouser, taxes instrument and analog devices website.
grateful for your help and comments.

now somehow I start to find many of them like: OP184, LT1366
